pkg/resources/provisioning.cattle.io/v1/cluster/mutator.go

```diff
@@ -133,13 +134,66 @@ func (m *ProvisioningClusterMutator) Admit(request *admission.Request) (*admissi
 		return response, nil
 	}
 
+	response, err = m.handleDynamicSchemaDrop(request, cluster)
+	if err != nil {
+		return nil, err
```

Comment:
Nit: would be good to wrap this so that we know which sub-function caused the error:

```suggestion
		return nil, fmt.Errorf("unable to evaluate dynamic schema drop, %w", err)
```